### PR TITLE
avoid calls to .address() when looking for endpoint protocol

### PR DESCRIPTION
**Pull request by aldenml**
I detected quite a few of these, but before going full in, I would like to have some opinions

### CODE REVIEW

**Comment by arvidn** [comment]
this looks like a reasonable approach.

**Comment by aldenml** [comment]
do you think it's better to merge this and do others PR(s)?

**Comment by arvidn** [comment]
what do you mean?

**Comment by aldenml** [comment]
that if you prefer that I do the other changes(more use of these new functions) in a separate PR, or just amend this one?

**Comment by arvidn** [comment]
I think amending this one makes sense

**Comment by aldenml** [comment]
OK, will do that

File: include/libtorrent/broadcast_socket.hpp [hunk 1]
```diff
@@ -53,6 +53,18 @@ namespace libtorrent {
 	TORRENT_EXTRA_EXPORT bool is_teredo(address const& addr);
 	TORRENT_EXTRA_EXPORT bool is_ip_address(std::string const& host);
 
+	// TODO: refactor these out too
```
Comment (arvidn) [inline include/libtorrent/broadcast_socket.hpp]: is this comment still relevant?

**Pull request merged**